src/context/auth.js

```diff
@@ -5,85 +5,98 @@ import Modal from "../components/modal";
-import jwt_decode from "jwt-decode"
+import jwt_decode from "jwt-decode";
 
-const AuthContext = createContext()
+const AuthContext = createContext();
 
 const AuthProvider = ({ children }) => {
-	const navigate = useNavigate()
-	const location = useLocation()
-	const [token, setToken] = useState(null)
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [token, setToken] = useState(null);
+  const [error, setError] = useState("");
 
-    useEffect(() => {
-        const storedToken = localStorage.getItem('token')
+  useEffect(() => {
+    const storedToken = localStorage.getItem("token");
 
-        if (storedToken) {
-            setToken(storedToken)
-            navigate(location.state?.from?.pathname || "/")
-        }
-    }, [location.state?.from?.pathname, navigate])
-
-	const handleLogin = async (email, password) => {
-		const res = await login(email, password)
-
-        if (!res.data.token) {
-            return navigate("/login")
-        }
-
-        localStorage.setItem('token', res.data.token)
+    if (storedToken) {
+      setToken(storedToken);
+      navigate(location.state?.from?.pathname || "/");
+    }
+  }, [location.state?.from?.pathname, navigate]);
 
-		setToken(res.token)
-		navigate(location.state?.from?.pathname || "/")
-	};
+  const handleLogin = async (email, password) => {
+    const res = await login(email, password);
 
-	const handleLogout = () => {
-        localStorage.removeItem('token')
-		setToken(null)
-	};
+    if (res.data.error) {
+      setError(res.data.error);
+    }
 
-    const handleRegister = async (email, password) => {
-        const res = await register(email, password)
-		setToken(res.data.token)
+    if (!res.data.token) {
+      return navigate("/login");
+    }
 
-        navigate("/verification")
+    localStorage.setItem("token", res.data.token);
+
+    setToken(res.data.token);
+    navigate(location.state?.from?.pathname || "/");
+    setError("");
+  };
+
+  const handleLogout = () => {
+    localStorage.removeItem("token");
+    setToken(null);
+    setError("");
+  };
+
+  const handleRegister = async (email, password) => {
+    const res = await register(email, password);
+    if (res.data.error) {
+      setError(res.data.error);
+    } else {
+      setToken(res.data.token);
+      navigate("/verification");
+      setError("");
     }
+  };
 
-    const handleCreateProfile = async (firstName, lastName, githubUrl, bio) => {
-        const { userId } = jwt_decode(token)
+  const handleCreateProfile = async (firstName, lastName, githubUrl, bio) => {
+    const { userId } = jwt_decode(token);
 
-        await createProfile(userId, firstName, lastName, githubUrl, bio)
+    await createProfile(userId, firstName, lastName, githubUrl, bio);
 
-        localStorage.setItem('token', token)
-        navigate('/')
-    }
+    localStorage.setItem("token", token); // Isn't it redundant, as the token is already stored in localStorage during login or registration?
```

Comment:
It might well be. You can either test it and remove this if it is redundant, or remove the comment and leave the code in